### PR TITLE
refactor: move the Primer API into its own package

### PR DESCRIPTION
**Pull request by dhess**
We would like to run evaluations in the browser, using GHC 9.6's support for Wasm and JavaScript targets. We won't need the API wrapper for that, nor type-checking, so this is the first step towards a more fine-grained set of packages in order to reduce the amount of code we need to compile for a Wasm/JavaScript target.

This is a pure refactor. No tests or functionality have been removed, nor any new functionality or tests added. We've simply shuffled things around a bit, and moved some additional code into `primer-testlib` so that it can be reused in `primer-api`'s tests.

### CODE REVIEW

**Comment by dhess** [comment]
Assuming there are no major objections to, or omissions in, this PR, could someone please add it to the merge queue once it's been reviewed and accepted?

File: primer/src/Primer/JSON.hs [hunk 1]
```diff
@@ -15,7 +15,7 @@ import Data.Aeson (
  )
 import Deriving.Aeson (CustomJSON (..), OmitNothingFields)
 
--- | A type for Primer API JSON encodings.
+-- | A type for Primer JSON encodings.x
```
Comment (brprice) [inline primer/src/Primer/JSON.hs:18]: Typo

File: primer-api/primer-api.cabal [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    LambdaCase
+    OverloadedStrings
+
+  ghc-options:
+    -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates
+    -Wcompat -Widentities -Wredundant-constraints
+    -Wmissing-deriving-strategies -fhide-source-paths
+
+  build-depends:
+    , aeson                        >=2.0      && <2.2
```
Comment (brprice) [inline primer-api/primer-api.cabal:36]: I wonder if any build-depends can be trimmed, either here or in the primer package now the api has been split?

**Review by brprice — APPROVED** [review]
I have only two minor comments, which I'll fix and then add this to the merge queue

File: primer-api/primer-api.cabal (hunk 2 above)
Comment (dhess) [reply primer-api/primer-api.cabal:36]: Yes, thanks, I meant to revisit that and forgot before I pushed this PR.

**Pull request merged**